### PR TITLE
MGMT-13292: Add the installed version of the OLM operators in the monitor operators response

### PR DESCRIPTION
**Pull request by eliorerz**
After https://github.com/openshift/assisted-service/pull/4961 we can merge the logic that brings the operator version version

### CODE REVIEW

File: src/assisted_installer_controller/operator_handler.go [hunk 1]
```diff
@@ -88,14 +88,18 @@ func (handler ClusterOperatorHandler) GetName() string { return handler.operator
...
 
-func (handler ClusterOperatorHandler) GetStatus() (models.OperatorStatus, string, error) {
+func (handler ClusterOperatorHandler) GetStatus() (models.OperatorStatus, string, string, error) {
 	co, err := handler.kc.GetClusterOperator(handler.operatorName)
 	if err != nil {
-		return "", "", err
+		return "", "", "", err
+	}
+	version := ""
+	if len(co.Status.Versions) > 0 {
```
Comment (gamli75) [inline src/assisted_installer_controller/operator_handler.go:97]: please also check that it's not nil

File: src/inventory_client/inventory_client.go [hunk 2]
```diff
@@ -348,6 +348,8 @@ func (c *inventoryClient) GetClusterMonitoredOLMOperators(ctx context.Context, c
 				operator.Name = "ocs"
 				operator.SubscriptionName = "ocs-operator"
 			}
+
+			operator.Version = "TODO" // todo
```
Comment (tsorya) [inline src/inventory_client/inventory_client.go:352]: what mean todo here?

File: src/assisted_installer_controller/operator_handler.go [hunk 3]
```diff
@@ -256,17 +260,17 @@ func (handler ClusterServiceVersionHandler) deleteFailedSubscriptionInstallPlans
...
+		return "", "", "", err
 	}
 	csv, err := handler.kc.GetCSV(handler.operator.Namespace, csvName)
 	if err != nil {
-		return "", "", err
+		return "", "", "", err
 	}
 	operatorStatus := utils.CsvStatusToOperatorStatus(string(csv.Status.Phase))
-	return operatorStatus, csv.Status.Message, nil
+	return operatorStatus, csv.Status.Message, csv.Spec.Version.String(), nil
```
Comment (tsorya) [inline src/assisted_installer_controller/operator_handler.go:273]: Maybe we should create struct for it? I mean 3 values to return not too much?

Comment (eliorerz) [reply src/assisted_installer_controller/operator_handler.go:273]: I agree, can we do it after that PR?

Comment (tsorya) [reply src/assisted_installer_controller/operator_handler.go:273]: yep, just lets not forgot it.

Comment (tsorya) [reply src/assisted_installer_controller/operator_handler.go:273]: Version can't be nil?

Comment (eliorerz) [reply src/assisted_installer_controller/operator_handler.go:273]: > yep, just lets not forgot it.

I'll open a task for it



> Version can't be nil?

Nope

File: src/inventory_client/inventory_client.go (hunk 2 above)
Comment (eliorerz) [reply src/inventory_client/inventory_client.go]: Thanks, removed

**Review by gamli75 — APPROVED** [review]

**Comment by eliorerz** [comment]
Results from CI job
```json
"monitored_operators": [
{
"cluster_id": "7fd2762b-74b7-4103-8de5-95a178249820",
"name": "cnv",
"namespace": "openshift-cnv",
"operator_type": "olm",
"status": "available",
"status_info": "install strategy completed with no errors",
"status_updated_at": "2023-02-16T02:28:11.527Z",
"subscription_name": "hco-operatorhub",
"timeout_seconds": 3600,
"version": "4.12.0"
},
{
"cluster_id": "7fd2762b-74b7-4103-8de5-95a178249820",
"name": "console",
"operator_type": "builtin",
"status": "available",
"status_info": "All is well",
"status_updated_at": "2023-02-16T02:13:35.886Z",
"timeout_seconds": 3600,
"version": "4.12.0-0.nightly-2023-02-15-091811"
},
{
"cluster_id": "7fd2762b-74b7-4103-8de5-95a178249820",
"name": "cvo",
"operator_type": "builtin",
"status": "available",
"status_info": "Done applying 4.12.0-0.nightly-2023-02-15-091811",
"status_updated_at": "2023-02-16T02:21:35.886Z",
"timeout_seconds": 3600,
"version": "4.12.0-0.nightly-2023-02-15-091811"
},
{
"cluster_id": "7fd2762b-74b7-4103-8de5-95a178249820",
"name": "lso",
"namespace": "openshift-local-storage",
"operator_type": "olm",
"status": "available",
"status_info": "install strategy completed with no errors",
"status_updated_at": "2023-02-16T02:26:11.552Z",
"subscription_name": "local-storage-operator",
"timeout_seconds": 4200,
"version": "4.12.0-202301042354"
}
]
```

**Pull request merged**